src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -448,20 +475,29 @@
         this)
       (a/stop [this]
         (shutdown! app-context)
-        this))))
+        this)
+      (a/restart [this]
+        (try
+          (run-lifecycle-fns app-context s/stop "stop" (reverse ordered-services))
```

Comment:
One thing that's different about this vs. what happens in `shutdown!` for an app stop is that if an exception bubbles from the `stop` lifecycle function called for one service, that exception is swallowed (but with an error message written to the log) but `stop` is still called on all of the other dependent services so that those services still have a chance to clean up their resources.  This implementation, however, would appear to cause the stop to abort after the first service's `stop` were to throw an exception.

I'm not sure if the `shutdown!` implementation is the best for all cases - e.g., if an aborted `stop` on one service were to cause the downstream `stop` calls on other services to behave abnormally for some reason.  I think it would be best, though, for a shutdown triggered from `stop` on the restart to behave the same as it would if triggered from app/stop.  So maybe the 'swallow the exception and move on to stopping the next service' approach is best for now.  @cprice404 - what do you think?